superset/db_engine_specs/druid.py

```diff
@@ -38,6 +52,14 @@ class DruidEngineSpec(BaseEngineSpec):  # pylint: disable=abstract-method
     allows_joins = False
     allows_subqueries = True
 
+    column_type_mappings = (
+        (
+            re.compile(r"^__time", re.IGNORECASE),
```

Comment:
It's actually the column name that is `__type`, not the type. So this needs to be picked up in `get_column_spec`, probably by adding the column name to the signature.